host.py

```diff
@@ -237,29 +247,30 @@ def ssh_connect(self, username: str, password: Optional[str] = None, rsa_path: s
             pw = PasswordLogin(self._hostname, username, password)
             self._logins.append(pw)
 
+        if discover_auth:
+            auto = AutoLogin(self._hostname, username)
+            self._logins.append(auto)
+
         self.ssh_connect_looped(self._logins)
 
-    def ssh_connect_looped(self, logins: list[Login]) -> None:
-        if len(logins) == 0:
-            raise Exception("No usable logins found")
+    def ssh_connect_looped(self, logins: list[Login], timeout: int = 360) -> None:
+        if not logins:
+            raise RuntimeError("No usable logins found")
 
-        login_details = ", ".join([str(login.vars()) for login in logins])
+        login_details = ", ".join([login.debug_details() for login in logins])
         logger.info(f"Attempting SSH connections on {self._hostname} with logins: {login_details}")
 
-        for _ in range(360):
+        for _ in range(timeout):
```

Comment:
`timeout` in this case seems a bad name for the variable. If it were a timeout, it should be in some understandable unit (e.g. seconds).

I kinda also dislike the pattern of repeating `timeout` times and sleeping for `time.sleep(10)` seconds, and assume that the total time is `timeout*10`. I'd prefer comparing the actually spent time against `time.monotonic()`. But ok. But then don't give the notion, that this would wait a predictable amount of time. The spent time is not well defined this way -- which may be fine, because the counter is chosen so large, that we don't expect it to be hit unless a problem exists. And then the exact time spent is not very important.

Reply:
That makes sense, I'll do it with `time.monotonic()`